### PR TITLE
feat: Implements message protection for the JDBC Queue

### LINKED ISSUE #4631: Better handling of large execution contexts
### Feature description

My company is currently working with a development partner for an internal ERP project. I have showcased and done training sessions with 90+ developers to onboard them on using Kestra for batch jobs / integration purposes. While I am fully familiar with the product, the people who I onboarded are not, and despite explaining to them the importance of minimizing data in the execution context, I believe Kestra should handle large execution contexts without crashing, rather than leaving it up to the developer to not accidentally store large execution contexts. 

I am creating this issue more as a general Epic, as this encompasses a lot of different features both on backend and frontend, and there is a lot that could be discussed on how to handle large execution contexts. 

The general issue with how Kestra currently handles large execution contexts is that 
both the webserver and executor will generally crash, and the DB state will block the executor from executing new tasks because of the large execution context stuck in the task queue. 

**Some things that I see could be improved in the webserver :** 

- The triggers column in the Executions tab will cause the Executions tab to completely freeze up and crash the webserver, if you use something like an SQL trigger and the trigger has a lot of rows / columns. Even if you only get top 100 rows, if all the executions are SQL triggers it'll try to load all the executions * trigger data, which will cause the Executions tab to load for a long time and / or crash the webserver. This can be worked around by triggering the flow by taking only the top 1 row with the relevant WHERE clause and then querying the data in the flow straight after, but the trigger popup in the Executions tab should still limit how much data is queried and shown (only show the top 15 fields as a preview). The below image is just showing which part of the UI I'm talking about, if I had an SQL with 10000 fields the Executions tab would attempt to load all of it for each execution. 
image

- Inputs in the Overview tab of the Flow execution should be paged, if there's a lot of input data the Overview tab will take a long time to load and / or crash the webserver. 

**Things that could be improved on the backend :** 

- Even while keeping in mind that I should minimize execution contexts, during development while testing I could make a mistake. For example, I could have an SQL Query task and query 200,000 rows and load that into the execution context by accident (without using the store attribute), which will most likely crash the executor and even create a crash loop as the executor will repeatedly try to load 200,000 rows into the execution context and keep crashing. 
       
Some possible solutions might be to add a configuration field either during deployment or on the admin page to add a max size limit (in bytes) of the execution context, so that the admin can configure the limit as allowed by the amount of java heap space that was provided during deployment. The executor could first verify the size of the execution context before attempting to load the data (I don't even know if this is possible, please just take this as a possible suggestion). If the size of the execution context exceeds that set by the administrator, the executor could signal the task to fail with an error message (I don't know if this is possible either). 

Regardless of how a solution could be implemented, the executor should handle cases with large execution contexts without crashing. My biggest worry is that a developer could not follow the best practices in creating a flow, accidentally or otherwise, and crash the deployment, which could be a huge deal in production.

### PR DESCRIPTION
**Pull request by loicmathieu**
Allow to set a limit in size for messages, when exceeded, the message will be refused by the Queue. The JDBC executor and worker has been updated to handle a too big message and fail the execution.

Fixes #4631

### CODE REVIEW

**Comment by loicmathieu** [comment]
- A message too big for a Worker Task will remove the task outputs and returned a failed worker task with a message in the UI. 
image
- A message too big for a WorkerTrigger will generate a log in the Kestra server log but not more (as failed trigger currently)
- A message too big for an execution will fail the execution with no error visible in the UI but a log in the Kestra server log

File: core/src/main/java/io/kestra/core/runners/RunContextLogger.java [hunk 1]
```diff
@@ -294,7 +295,13 @@ protected void append(ILoggingEvent e) {
             e = this.transform(e);
 
             logEntries(e, logEntry)
-                .forEach(logQueue::emitAsync);
+                .forEach(log -> {
+                    try {
+                        logQueue.emitAsync(log);
+                    } catch (QueueException ex) {
+                        // silently do nothing
```
Comment (Skraye) [inline core/src/main/java/io/kestra/core/runners/RunContextLogger.java:302]: I think a catch should never do nothing, at least log something

Comment (loicmathieu) [reply core/src/main/java/io/kestra/core/runners/RunContextLogger.java:302]: In 99% of cases you"re right, but here is the exception to this rule, if we cannot log (which will never happen), logging in the server log is not necessary as the log would already have been emitted to it (and moreover, in this case something really wrong is happening so there should already be thousands of logs in the server log.

If it's very important to you I can add a debug log thought ...

**Comment by anna-geller** [comment]
@loicmathieu can you document where users can set that message size limit in the configuration? I couldn't find any docs for this in the PR

**Pull request merged**